csm/conf/upgrade.py

```diff
@@ -28,5 +30,114 @@ def __init__(self):
 
     async def execute(self, command):
         Log.info("Perform upgrade for csm_setup")
-        # TODO: Implement upgrade logic
+        try:
+            Log.info("Loading Url into conf store.")
+            Conf.load(const.CONSUMER_INDEX, command.options.get(const.CONFIG_URL))
+
+            Log.info("Loading current csm configurations.")
+            self.load_csm_config_indices()
+
+            Log.info("Loading default csm configurations.")
+            self.load_default_config()
+        except KvError as e:
+            Log.error(f"Configuration Loading Failed {e}")
+            raise CsmSetupError("Could Not Load Url Provided in Kv Store.")
+
+        services = command.options.get("services")
+        if ',' in services:
+            services = services.split(",")
+        elif 'all' in services:
```

Comment:
This is not required. Web and CLI are not part of CSM Agent now. You should throw error if other than all or agent is given.
All = agent only

Reply:
Resolved.